### DISCUSSION TITLE
Logs suddenly full of connection errrors

### DISCUSSION THREAD

**Discussion by bmrowe** [teslamate-org/teslamate#2375] [Q&A]
I've got an updated setup that has been working fine. However, I noticed a 'health check failed' error on the homepage today and went in the logs. The logs are full of:

> 2022-01-26 09:11:29.413 [info] GET /settings
> 2022-01-26 09:11:29.420 [info] Sent 200 in 6ms
> 2022-01-26 09:11:43.218 [warning] GET https://owner-api.teslamotors.com/api/1/vehicles/[redacted]/vehicle_data -> 503 (321.069 ms)
> 
> 2022-01-26 09:11:43.218 [warning] TeslaApi.Error / "upstream connect error or disconnect/reset before headers. reset reason: connection failure"
> 
> 2022-01-26 09:11:43.218 car_id=1 [error] Error / :unknown
> 
> 2022-01-26 09:11:47.548 [info] GET /
> 2022-01-26 09:11:47.551 [info] Sent 200 in 2ms
> 2022-01-26 09:12:30.324 car_id=1 [info] Start / :online
> 2022-01-26 09:12:30.380 car_id=1 [info] Connecting ...
> 2022-01-26 09:15:34.145 car_id=1 [info] Suspending logging
> 2022-01-26 09:16:32.046 [notice] :alarm_handler: {:clear, TeslaMate.Vehicles.Vehicle_1_api_error}

Ideas?

**Comment by cwanja** [comment]
This typically happens when the connection from TeslaMate to Tesla's API failed. In this instance, a 503 - which typically means a service is unavailable. I have seen these more frequently in my own installation (not specific to TeslaMate, I think it is more Tesla hardware related) and it resolves itself within about one to three minutes. You can validate this error by opening the Tesla app and seeing what it says in the upper left hand corner. I imagine it will say `503 Service Unavailable`.

**Reply by cwanja** [reply]
Similar to this. 
![F9D8D3A9-D06E-4370-8DE5-A7B3AB4A0678](https://user-images.githubusercontent.com/54645561/151678131-30dbf376-13e1-477c-a0ad-cbb3bed54153.jpeg)

**Comment by dsgdesign** [comment]
Hi teslamate dev community and users,

This old thread is the closest I have found that corresponds to my issue. Maybe someone can shed some light on what is going on since about over a week now. Here is a recent excerpt of my teslamate log (seen repeatedly on the log for over a week):

```
teslamate-1  | 2026-02-19 20:35:50.439 [info] Scheduling token refresh in 6 h
teslamate-1  | 2026-02-20 02:35:50.440 [info] Refreshing access token ...
teslamate-1  | 2026-02-20 02:35:50.741 [info] POST https://auth.tesla.com/oauth2/v3/token -> 200 (299.868 ms)
teslamate-1  | 2026-02-20 02:35:50.753 [info] Scheduling token refresh in 6 h
teslamate-1  | 2026-02-20 05:40:59.011 [warning] GET https://owner-api.teslamotors.com/api/1/vehicles/149XXXXXXXXXXXXX -> 503 (184.934 ms)
teslamate-1  | 2026-02-20 05:40:59.011 [warning] TeslaApi.Error / "upstream connect error or disconnect/reset before headers. reset reason: overflow"
teslamate-1  | 2026-02-20 05:40:59.011 car_id=1 [error] Error / :unknown
```

Environment:

Teslamate version 2.2.0
OS: Ubuntu Budgie 22.04.5 LTS aarch64 
Host: Raspberry Pi 4 Model B Rev 1.4 
Kernel: 5.15.0-1093-raspi
Client: Docker Engine - Community
Version:           29.2.1
API version:       1.53

Tesla model 3 vehicle on version 2026.2.3

I have checked connectivity, DNS, etc. The above response to the GET request as being 503 Service unavailable does make sense, but I have not seen the iOS Tesla app showing the same message and since the issue keeps showing up I do not see it resolving itself. All the Grafana info, location of the car, and its past and present states appear to be working. Thanks again to the devs for all the work. Now on my fourth year running Teslamate!

**Reply by JakobLichterfeld** [reply]
Do you live in a geo-blocked location?

**Reply by dsgdesign** [reply]
I'm located in Canada, which is not geo-blocked, and I am not behind a VPN. However, my Teslamate Raspberry Pi  now sits behind a new firewall installation and, as per Teslamate docs (FAQ) I have applied the following policies:

The following flows must be authorized (egress traffic and DNS resolution):

HTTPS (TCP/443)
auth.tesla.com
owner-api.teslamotors.com
streaming.vn.teslamotors.com
nominatim.openstreetmap.org

HTTP (TCP/80)
step.esa.int

I will look into my firewall logs, but I doubt this is where I will find an answer. Thanks again for your input.
